src/librustdoc/core.rs

```diff
@@ -570,6 +580,73 @@ pub fn run_core(options: RustdocOptions) -> (clean::Crate, RenderInfo, RenderOpt
     })
 }
 
+use rustc_hir::def::Res;
+use rustc_hir::{
+    intravisit::{NestedVisitorMap, Visitor},
+    Path,
+};
+use rustc_middle::hir::map::Map;
+
+/*
+thread_local!(static DEFAULT_TYPECK: for<'tcx> fn(rustc_middle::ty::TyCtxt<'tcx>, rustc_span::def_id::LocalDefId) -> &'tcx rustc_middle::ty::TypeckTables<'tcx> = {
```

Comment:
Maybe remove this code if it's not used anymore?

Reply:
This is waiting on https://github.com/rust-lang/rust/pull/74070 and then it should be uncommented and the current code updated.

Reply:
Right now it gives the following error:
```rust
error[E0308]: mismatched types
   --> src/librustdoc/core.rs:594:5
    |
591 | thread_local!(static DEFAULT_TYPECK: for<'tcx> fn(TyCtxt<'tcx>, LocalDefId) -> &rustc_middle::ty::TypeckTables<'tcx> = {
    |                                      ------------------------------------------------------------------------------- expected `for<'tcx> fn(rustc_middle::ty::TyCtxt<'tcx>, rustc_span::def_id::LocalDefId) -> &'tcx rustc_middle::ty::TypeckTables<'tcx>` because of return type
...
594 |     providers.typeck_tables_of
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected concrete lifetime, found bound lifetime parameter 'tcx
```